logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/UserEventReceiver.kt

```diff
@@ -85,7 +96,20 @@ internal class UserEventReceiverImpl internal constructor(
     private suspend fun handleNewConnection(event: Event.User.NewConnection): Either<CoreFailure, Unit> =
         connectionRepository.insertConnectionFromEvent(event)
             .flatMap {
-                resolveActiveOneOnOneConversationUponConnectionAccepted(event.connection)
+                if (event.connection.status != ConnectionState.ACCEPTED) {
+                    return@flatMap Either.Right(Unit)
+                }
+
+                if (event.live) {
+                    resolveActiveOneOnOneScope.launch {
+                        kaliumLogger.d("Delay resolving active one-on-one since we are live to avoid race conditions")
+                        delay(3.seconds)
+                        oneOnOneResolver.resolveOneOnOneConversationWithUserId(event.connection.qualifiedToId).map { }
+                    }
```

Comment:
Just to confirm.

The idea here is that during these 3 seconds, the protocol might have been updated already. So when it calls `resolveOneOnOneConversationWithUserId`, the resolver will notice that it is already MLS, and shortcircuit successfully.
And checking if `live` or not makes a difference because if it's not live, we can assume that the migration was already solved and we'll know about it shortly after in the coming events.
Correct?

Makes sense to me. I wonder if adding a small jitter would be nice.

Something like $delay = 3 + [0..3] seconds$

Attempting to reduce the chance of clients racing to migrate.

Probably not worth it.

Reply:
The idea is that during 3 seconds we will in 99% of the cases receive a welcome message which establishes the MLS group. If there's no errors it's the client which accepts the connection which also establish the group.

Checking if we are `live` or not is to avoid introducing a delays when catching up. But I realised now that we need to post pone resolving the 1-1 also in the non-live case since otherwise we would discard the welcome message.

I'll fix this.